sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -51,7 +51,12 @@ internal class LazyLoadingReadOnlyStream<TRequestConditions, TProperties> : Stre
         /// <summary>
         /// If we are allowing the blob to be modifed while we read it.
         /// </summary>
-        private bool _allowBlobModifications;
+        private readonly bool _allowBlobModifications;
+
+        /// <summary>
+        /// Indicated the user has called Seek() since the last Read() call, and the new position is outside _buffer.
+        /// </summary>
+        private bool _seeked;
```

Comment:
Nit - `_seeked` is kind of a funky name that describes more about the history of actions rather than the current state of the object.  What about a name like `_bufferInvalidated` to make code like `if (_bufferPosition == 0 || _bufferPosition == _bufferLength || _bufferInvalidated)` read a little easier?

Reply:
fixed.